client/me/purchases/purchase-item/index.tsx

```diff
@@ -626,24 +629,40 @@ export function PurchaseItemPaymentMethod( {
 		);
 	}
 
+	const goToAddPaymentMethod = ( e: React.MouseEvent< HTMLDivElement >, siteId: number ) => {
+		e.preventDefault();
+		e.stopPropagation();
+
+		page( `/me/purchases/add-payment-method/${ siteId }/payment-method/add` );
+	};
+
 	if (
 		purchase.isAutoRenewEnabled &&
-		! hasPaymentMethod( purchase ) &&
+		( ! hasPaymentMethod( purchase ) || isPaidWithCredits( purchase ) ) &&
 		! isPartnerPurchase( purchase ) &&
 		! isAkismetFreeProduct( purchase )
 	) {
 		return (
 			<div className="purchase-item__no-payment-method">
-				<Icon icon={ warningIcon } />
-				<span>{ translate( 'You don’t have a payment method to renew this subscription' ) }</span>
+				{ ! isDisconnectedSite && (
+					<Button
+						variant="primary"
+						size="compact"
+						onClick={ ( e: React.MouseEvent< HTMLDivElement > ) =>
+							goToAddPaymentMethod( e, purchase.id )
+						}
```

Comment:
This could simply be

```suggestion
						onClick={ () =>
							page( `/me/purchases/add-payment-method/${ siteId }/payment-method/add` )
						}
```

Reply:
Thanks for the suggestion, but I need to stop propagation, so the committed approach makes more sense.